### DISCUSSION TITLE
How to get data of indirect relation in API rules of Admin UI and filter based on relation column?

### DISCUSSION THREAD

**Discussion by samimhamza-pgl** [pocketbase/pocketbase#2275] [Q&A] [answered]
I have a permissions collection with id, name, created, update columns and user_permissions collection with id, user_id, permission_id, created, updated columns. Now I want to restrict pocketbase APIs through permissions that each user has.
I intentionally added user_permissions collection instead of just adding permissions column to users  because I don't want to directly show permissions when fetching users collection.
How can I restrict collections API rules based on user_permissions collection and permissions collection?

**Comment by ganigeorgiev** [comment, marked as answer]
Indirect relations filter and sort is planned for the next v0.15/v0.16 release.

As a more general solution you ca use the `@collection.*` filter. It performs SQL `JOIN` with the specified collection. Depending from where you want to check the permissions, this could look something like:

```js
@request.auth.id != "" &&
@collection.user_permissions.user_id ?= @request.auth.id &&
@collection.user_permissions.permission_id.name ?= "somePermission"
```

But I'd recommend instead of heaving a dedicated `user_permissions` collection, to define a `permissions` multiple `relation` field in the `users` collection that will point to the `permissions` collection. The user will see only the `permissions` ids and you still will have the option to restrict the list/expand to prevent listing the permission names if that's your concern, but you'll also have easier access to the permissions through the filters. For example the previous rile in this case could be rewritten to:

```js
@request.auth.id != "" &&
@request.auth.permissions.names ?= "somePermission"
```

**Reply by samimhamza-pgl** [reply]
ok thanks, it works perfectly
